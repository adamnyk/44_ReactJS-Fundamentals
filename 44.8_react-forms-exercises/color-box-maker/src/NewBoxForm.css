.NewBoxForm {
    text-align: center;
    margin-bottom: 20px;
}

.NewBoxForm-input-group {
    margin: 10px;
}

.NewBoxForm input {
    margin-left: 5px;
}

.NewBoxForm span {
    margin-left: 10px;
    font-family: monospace;
}