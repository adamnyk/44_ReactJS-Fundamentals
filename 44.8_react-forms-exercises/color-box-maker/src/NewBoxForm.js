import { useState } from "react";
import "./NewBoxForm.css";

const NewBoxForm = ({addBox}) => {

    const INITIAL_STATE = {color:"black", height:50, width:50}
    const [formData, setFormData] = useState(INITIAL_STATE)

    const handleChange = (e) => {
        const { name, value } = e.target
        setFormData(formData => ({...formData, [name]:value}))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addBox(formData);
        // setFormData(INITIAL_STATE) //use to reset form after submit
    }
    

	return (
        <form
            className="NewBoxForm"
            onSubmit={handleSubmit}
        >
			<div className="NewBoxForm-input-group">
                <label htmlFor="color">Color:</label>
                <input 
                                name="color" 
                                id="color" 
                                type="color" 
                                value={formData.color}
                                onChange={handleChange}
                />
                                <span>{formData.color}</span>

            </div>

            <div className="NewBoxForm-input-group">
                <label htmlFor="height">Height:</label>
                    <input
                        name="height"
                        id="height"
                        type="range"
                        min={50}
                        max={500}
                        step={25}
                        value={formData.height}
                        onChange={handleChange}
                        />
                <span>{formData.height}px</span>
            </div>

            <div className="NewBoxForm-input-group">
                <label htmlFor="width">Width:</label>
                    <input
                        name="width"
                        id="width"
                        type="range"
                        min={50}
                        max={500}
                        step={25}
                        value={formData.width}
                        onChange={handleChange}
                    />
                <span>{formData.width}px</span>
            </div>

            <button>Make box</button>
		</form>
	);
};

export default NewBoxForm;
