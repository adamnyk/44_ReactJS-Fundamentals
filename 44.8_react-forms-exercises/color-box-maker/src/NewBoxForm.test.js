import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it("Box renders", () => {
	render(<NewBoxForm />);
});

it("matches snapshot", function () {
	const { asFragment } = render(<NewBoxForm />);
	expect(asFragment()).toMatchSnapshot();
});
