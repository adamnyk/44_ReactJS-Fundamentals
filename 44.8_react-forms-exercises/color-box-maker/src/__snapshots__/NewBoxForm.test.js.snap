// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <div
      class="NewBoxForm-input-group"
    >
      <label
        for="color"
      >
        Color:
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value="black"
      />
      <span>
        black
      </span>
    </div>
    <div
      class="NewBoxForm-input-group"
    >
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        max="500"
        min="50"
        name="height"
        step="25"
        type="range"
        value="50"
      />
      <span>
        50px
      </span>
    </div>
    <div
      class="NewBoxForm-input-group"
    >
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        max="500"
        min="50"
        name="width"
        step="25"
        type="range"
        value="50"
      />
      <span>
        50px
      </span>
    </div>
    <button>
      Make box
    </button>
  </form>
</DocumentFragment>
`;
